ui/src/app/views/workspace/component.spec.ts

```diff
@@ -0,0 +1,109 @@
+import {Component, DebugElement} from '@angular/core';
+import {TestBed, async, fakeAsync, ComponentFixture} from '@angular/core/testing';
+import {Title, By} from '@angular/platform-browser';
+import {ActivatedRoute, UrlSegment} from '@angular/router';
+import {RouterTestingModule} from '@angular/router/testing';
+import {WorkspaceComponent} from 'app/views/workspace/component';
+import {updateAndTick, simulateInput} from 'testing/test-helpers';
+import {CohortsServiceStub} from 'testing/stubs/cohort-service-stub';
+import {CohortsService} from 'generated';
+import {UserService} from 'app/services/user.service';
+import {RepositoryService} from 'app/services/repository.service';
+
+class Context {
+  fixture: ComponentFixture<WorkspaceComponent>;
+  cohortsService: CohortsService;
+  userService: UserService;
+  repositoryService: RepositoryService;
+  route: UrlSegment[];
+  cohortsTableRows: DebugElement[];
+  notebookTableRows: DebugElement[];
+  cdrText: DebugElement;
+  workspaceDescription: DebugElement;
+  loggedOutMessage: DebugElement;
+
+  constructor(testBed: typeof TestBed) {
+    this.fixture = testBed.createComponent(WorkspaceComponent);
+    this.cohortsService = this.fixture.debugElement.injector.get(CohortsService);
+    this.userService = this.fixture.debugElement.injector.get(UserService);
+    this.repositoryService = this.fixture.debugElement.injector.get(RepositoryService);
+    this.route = this.fixture.debugElement.injector.get(ActivatedRoute).snapshot.url;
+    updateAndTick(this.fixture);
+    updateAndTick(this.fixture);
+    this.readPageData();
+  }
+
+  readPageData() {
+    this.cohortsTableRows = this.fixture.debugElement.queryAll(By.css('.cohort-table-row'));
+    this.notebookTableRows = this.fixture.debugElement.queryAll(By.css('.notebook-table-row'));
+    this.cdrText = this.fixture.debugElement.query(By.css('#cdr-text'));
+    this.workspaceDescription = this.fixture.debugElement.query(By.css('.description-text'));
+    this.loggedOutMessage = this.fixture.debugElement.query(By.css('#logged-out-message'));
+  }
+}
+
+const activatedRouteStub  = {
+  snapshot: {
+    url: [
+      {path: 'workspace'},
+      {path: WorkspaceComponent.DEFAULT_WORKSPACE_NS},
+      {path: WorkspaceComponent.DEFAULT_WORKSPACE_ID}
+    ]
+  }
+};
+
+describe('WorkspaceComponent', () => {
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        RouterTestingModule
+      ],
+      declarations: [
+        WorkspaceComponent
+      ],
+      providers: [
+        { provide: CohortsService, useValue: new CohortsServiceStub() },
+        { provide: UserService, useValue: new UserService() },
+        { provide: RepositoryService, useValue: new RepositoryService() },
+        { provide: ActivatedRoute, useValue: activatedRouteStub }
+      ] }).compileComponents();
+  }));
+
+
+  it('displays correct information in default workspace', fakeAsync(() => {
```

Comment:
For now I think it makes sense to check the row count in this test, although in the future it may make sense to add a test that reads the names. (I could write that now?)
I have changed the cohorts test to read from the service and compare that the same number are displayed as are read from the service.
However, for notebooks, currently that is hardcoded and we don't actually have a notebooks service yet. I'm inclined to build out that test when we actually are reading in notebooks, once we have a service to stub.